docs/persistent_compilation_cache.md

```diff
@@ -67,8 +67,111 @@ Cloud Storage (GCS) bucket. We recommend the following configuration:
 Assuming that `gs://jax-cache` is the GCS bucket, set `cache-location` as
 follows:
 
-```
+```python
 import jax
 
 jax.config.update("jax_compilation_cache_dir", "gs://jax-cache")
 ```
+
+## How it works
+
+The JAX compilation cache works by hashing a number of parameters to create a signature for a compiled function these are:
+
+* The computation performed by the function captured by the non-optimized HLO of the JAX function being hashed
+* The Jaxlib version 
+* Relevant XLA compilation flags
+* Device configuration captured in general, by the number of devices and the topology of the devices. Currently for GPUs, the topology only contains a string representation of the GPU name
+* Compression algorithm used to compress the compiled executable
+* Any custom hooks
+
+When the signature for a function created using the parameters above matches that of a compiled function in the persistent cache, the function will not be compiled but just read and deserialized from the persistent cache. 
+
+### Compile time dependent caching
```

Comment:
Also need to mention `jax_persistent_cache_min_entry_size_bytes`.

Besides, this topic does not seem substantial enough to be a standalone section.